src/lib/lite_base/protocol_state.ml

```diff
@@ -1,18 +1,48 @@
 open Fold_lib
 
 module Body = struct
-  type t =
-    {blockchain_state: Blockchain_state.t; consensus_state: Consensus_state.t}
-  [@@deriving eq, bin_io, sexp]
+  module Stable = struct
+    module V1 = struct
+      module T = struct
+        type t =
+          { blockchain_state: Blockchain_state.Stable.V1.t
+          ; consensus_state: Consensus_state.Stable.V1.t }
+        [@@deriving eq, bin_io, sexp, version]
+      end
+
+      include T
+    end
+
+    module Latest = V1
+  end
+
+  type t = Stable.Latest.t =
+    { blockchain_state: Blockchain_state.Stable.V1.t
+    ; consensus_state: Consensus_state.Stable.V1.t }
+  [@@deriving eq, sexp]
 
   let fold {blockchain_state; consensus_state} =
     let open Fold in
     Blockchain_state.fold blockchain_state
     +> Consensus_state.fold consensus_state
 end
 
-type t = {previous_state_hash: Pedersen.Digest.t; body: Body.t}
-[@@deriving eq, bin_io, sexp]
+module Stable = struct
+  module V1 = struct
+    module T = struct
+      type t = {previous_state_hash: Pedersen.Digest.t; body: Body.Stable.V1.t}
```

Comment:
I think you can use `Pedersen.Digest.Stable.V1.t` here, and remove the `asserted`.

Reply:
Pedersen.Digest is not stable yet.